packages/SwingSet/src/types-internal.js

```diff
@@ -0,0 +1,15 @@
+import './types-exported';
```

Comment:
This is a real ES import, so it needs an extension.

```suggestion
import './types-exported.js';
```

But that bring the question why this is needed? Is it for bringing in the ambient types from `./types.js`? In which case why not go straight for the source of ambientness in the first place?

```suggestion
import './types.js';
```

Reply:
Argh, I know I typed `./types-exported.js`, but something auto-corrected the extension away. I'll add it back.

It looks like `import ./types.js` works too.

With those changes, we've got:
* exported.js : import './src/vats/timer/types.js'
* src/index.js: export * from './types-exported.js'
* src/types-exported.js: import './types.js', export {}, some typedefs
* src/types.js: only typedefs, no JS imports or exports, TS import(types-exported) and (@endo/marshal) and ./vatKeeper, ./kernelKeeper, others
* src/types-internal.js: import './types.js', export {}, some typedefs
* kernel.js: TS import(types-internal)

If I understand it right, `src/types.js` lacks a JS `import` or `export`, so everything defined there is added to the ambient type set (including things like `@typedef { import('./types-exported.js').BundleID } BundleID`). Those are necessary for now, but we want to move away from that and make everything be imported/exported properly. So the goal is to move things out of `types.js` until it's empty, and we can rename everything to be shorter.

`src/index.js` re-exports everything from `types-exported.js`. This is where we want to go: other packages should reference types with `{ import('@agoric/swingset-vat').BundleID }`, and we should move externally-visible types from `types.js` to `types-exported.js`. This includes the API surface of `controller`, but also things like `vatAdminService` and the timer service. Today's addition is the `options` bag that `E(vatAdminService).createVat(bundlecap, options)` takes.

`types-internal.js` is where I'm moving things from `types.js` that are only for internal use. For example `kernel.js` now does:

```js
   * @typedef { import('../types-internal.js').InternalDynamicVatOptions } InternalDynamicVatOptions
```

I thought about defining that in `types.js` and having `kernel.js` import from there, but TS complains that `types.js` is "not a module" (which I assume is because it lacks a JS `export {}`). I also assume that adding `export {}` to fix that would disqualify `types.js` from contributing to the ambient type definitions, which would break the entire world (since it still depends on those ambient types).

If that all sounds correct, I'll stick with adding new things to `types-internal.js` (so I'm not accidentally adding them to the ambient set, which might hide things we add without also importing). But since `types-internal.js` does a `import './types.js'`, I should expect `@typedef { import('../types-internal.js').Foo } Foo` to be able to pick up anything from `types.js`.

In the long run, once we've drained `types.js`, I'll want to rename `types-exported.js` and `types-internal.js`. I don't know which one should win the short name, but hopefully by the time swingset gets there, somebody will have figured out a good pattern in some other package and I can just copy it.